src/Arcus.Messaging.Tests.Integration/Fixture/WorkerOptions.cs

```diff
@@ -78,30 +79,33 @@ public WorkerOptions ConfigureSerilog(Action<LoggerConfiguration> configure)
         internal void ApplyOptions(IHostBuilder hostBuilder)
         {
             Guard.NotNull(hostBuilder, nameof(hostBuilder), "Requires a host builder instance to apply the worker options to");
-            
-            hostBuilder.ConfigureAppConfiguration(config => config.AddInMemoryCollection(Configuration))
+
+            LoggerConfiguration config =
+                new LoggerConfiguration()
+                    .MinimumLevel.Verbose()
+                    .Enrich.FromLogContext();
+
+            if (_outputWriter != null)
+            {
+                config.WriteTo.XunitTestLogging(_outputWriter);
+            }
+
+            foreach (Action<LoggerConfiguration> configure in _additionalSerilogConfigOptions)
+            {
+                configure(config);
+            }
+
+            Log.Logger = config.CreateLogger();
+
+            hostBuilder.ConfigureAppConfiguration(builder => builder.AddInMemoryCollection(Configuration))
                        .ConfigureServices(services =>
                        {
                            foreach (ServiceDescriptor service in Services)
                            {
                                services.Add(service);
                            }
                        })
-                       .UseSerilog((context, config) =>
-                       {
-                           config.MinimumLevel.Verbose()
-                                 .Enrich.FromLogContext();
-
-                           if (_outputWriter != null)
-                           {
-                               config.WriteTo.XunitTestLogging(_outputWriter);
-                           }
-
-                           foreach (Action<LoggerConfiguration> configure in _additionalSerilogConfigOptions)
-                           {
-                               configure(config);
-                           }
-                       });
+                       .UseSerilog(Log.Logger);
```

Comment:
After some investigation and step-by-step pinpointing, I have reduced the space of the problem to Serilog and its registration. I changed it to this, more explicit, way which finally made the test work again.

Reply:
Weird.  Do you have any idea what the exact reason was why it failed ?

Reply:
The test has two Serilog sinks; 1 to stream logs to the test output, 1 to save to an in-memory App Insights sink to later assert on the telemetry.
The test in particular sometimes failed in the past (only remotely), but recently, it went smoothly again, until now where it failed constantly. The exact reason, I don’t know. It has to be environment-specific, I’m thinking about VM’s and versions, even. The test never failed to run locally, even together with other tests.

Reply:
Weird, but good that it's working again :)